### DISCUSSION TITLE
TypeScript output

### DISCUSSION THREAD

**Discussion by lilactown** [clavascript/clavascript#307] [Ideas]
I recently had a bug in my brain that led me to coming up with an imaginary CLJS syntax that compiled to TypeScript annotations. I wrote it down in a random gist here: https://gist.github.com/lilactown/e924ec5dfe801a596ffd02477ddfd55f

It takes advantage of metadata to annotate a form, similar to how type annotations are done in Clojure today, but extends it to allow passing arbitrary forms as the tag which then the compiler would parse produce the accompanying TypeScript output. If the Clojure(Script) compiler was extended to allow this too (as a no-op), the code would remain valid ClojureScript consumable by other tools as well.

As long as the reader supports this, I don't see any complications other than a [small matter of programming](https://en.wikipedia.org/wiki/Small_matter_of_programming) to produce the correct TS output.

**Comment by borkdude** [comment]
Related: https://github.com/squint-cljs/squint/issues/434

**Reply by borkdude** [reply]
Progress: https://twitter.com/borkdude/status/1735989060823404730/photo/1
